### PR TITLE
IOS-7435 Add title spec

### PR DESCRIPTION
**Pull request by jmpg93**
🎟️ Jira ticket

[IOS-7435](https://jira.tid.es/browse/IOS-7435) 

🥅 What's the goal?
- Implement the new Title component.
- Update fonts present >= 5 to support different weights depending on the brand.

🚧 How do we do it?
- To implement the new `Title` component I've reused the old `SectionTitleView` which is deprecated ([spec](https://www.figma.com/file/yQRVCOmRQNM98bnXndLPxK/Title-Specs?node-id=0%3A1)).
- I'm added a new configuration to the `MisticaConfig` to decide whether or not text preset above 5 should use `bold` instead of `light`. Only Movistar uses this new config.
- I've fixed tests so I can properly install and use fonts in tests.
- Every component using preset needs new screenshots, that's why there are a lot of files involved.

### CODE REVIEW

File: Mistica/Source/Components/EmptyState/Internals/EmptyStateMessagesContent.swift [hunk 1]
```diff
@@ -98,7 +98,7 @@ extension EmptyStateMessagesContent {
         titleTopSpacing = titleSpacing
         let descriptionSpacing: CGFloat = isCard ? Constants.descriptionTopSpacingCard : Constants.descriptionTopSpacingDefault
         descriptionTitleTopSpacing = descriptionSpacing
-        titleLabel.font = isCard ? .textPreset4(weight: .regular) : .textPreset6(weight: .light)
+        titleLabel.font = isCard ? .textPreset4(weight: .regular) : .textPreset6()
```
Comment (jmpg93) [inline Mistica/Source/Components/EmptyState/Internals/EmptyStateMessagesContent.swift:101]: You can no longer decide `weight` on text presets above 5.

**Comment by yceballost** [comment]
Title is not aligned to the link, is it possible to align link to the title baseline?
image

**Comment by jmpg93** [comment]
> Title is not aligned to the link, is it possible to align link to the title baseline? image
> 
> A little mistake, in the example screen you can read "Title1 with .." but this one is Title2

Wups, I've fixed it, I'm generating a new version of the catalog.

demo

**Review by yceballost — APPROVED** [review]
🤟✨

**Review by cbarcom — APPROVED** [review]

**Review by DavidMarinCalleja — APPROVED** [review]
Good job!

**Pull request merged**